server/graphql/invoice/src/mutations/outbound_return/update.rs

```diff
@@ -0,0 +1,121 @@
+use async_graphql::*;
+
+use graphql_core::{
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use graphql_types::types::InvoiceNode;
+use service::invoice::outbound_return::update::{
+    UpdateOutboundReturn as ServiceInput, UpdateOutboundReturnError as ServiceError,
+};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    invoice::outbound_return::update::UpdateOutboundReturnStatus,
+};
+
+use super::insert::OutboundReturnLineInput;
+
+#[derive(InputObject)]
+#[graphql(name = "UpdateOutboundReturnInput")]
+pub struct UpdateInput {
+    pub id: String,
+    // supplier_id: String,
+    status: Option<UpdateOutboundReturnStatusInput>,
+    outbound_return_lines: Vec<OutboundReturnLineInput>,
+}
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq, Debug)]
+pub enum UpdateOutboundReturnStatusInput {
+    Allocated,
+    Picked,
+    Shipped,
+}
+
+#[derive(Union)]
+#[graphql(name = "UpdateOutboundReturnResponse")]
+pub enum UpdateResponse {
+    Response(InvoiceNode),
+}
+
+pub fn update(ctx: &Context<'_>, store_id: &str, input: UpdateInput) -> Result<UpdateResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            // resource: Resource::MutateOutboundReturn, // TODO
+            resource: Resource::MutateInboundShipment,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.to_string(), user.user_id)?;
+
+    let result = service_provider
+        .invoice_service
+        .update_outbound_return(&service_context, input.to_domain());
+
+    match result {
+        Ok(outbound_return) => Ok(UpdateResponse::Response(InvoiceNode::from_domain(
+            outbound_return,
+        ))),
+        Err(err) => map_error(err),
+    }
+}
+
+fn map_error(error: ServiceError) -> Result<UpdateResponse> {
+    use StandardGraphqlError::*;
+    let formatted_error = format!("{:#?}", error);
+
+    let graphql_error = match error {
+        // TODO - are any of the below actually structured errors?
+        // ServiceError::OtherPartyNotVisible => {
+        //     return Ok(UpdateErrorInterface::OtherPartyNotVisible(
+        //         OtherPartyNotVisible,
+        //     ))
+        // }
+
+        // Standard Graphql Errors
+        ServiceError::NotAnOutboundReturn
+        | ServiceError::ReturnDoesNotBelongToCurrentStore
+        | ServiceError::ReturnIsNotEditable
+        | ServiceError::ReturnDoesNotExist => BadUserInput(formatted_error),
```

Comment:
Are any of these actually structured errors? I've seen ones like these as both structured and standard, but going on the "can the frontend prevent this" I think mostly these would be considered standard errors?

Except perhaps `ReturnIsNotEditable`... someone else could update the status while we're editing? Oh and then perhaps same goes for `ReturnDoesNotExist` - they could have deleted it?